File: public/brand/android-theme/flutter/pocketgull_theme.dart
import 'package:flutter/material.dart';

/// 📜 PocketGull Washi Rice Paper ThemeData (Material 3)
final ThemeData pocketgullWashiTheme = ThemeData(
  useMaterial3: true,
  brightness: Brightness.light,
  scaffoldBackgroundColor: const Color(0xFFFAF8F2),
  colorScheme: const ColorScheme.light(
    primary: Color(0xFF0D9488),       // Celadon Bamboo Teal
    onPrimary: Colors.white,
    secondary: Color(0xFFD97706),     // Persimmon Amber
    onSecondary: Colors.white,
    tertiary: Color(0xFF0284C7),      // Medical Indigo
    surface: Color(0xFFF5F2E9),
    onSurface: Color(0xFF292524),     // Sumi Charcoal Ink
    outline: Color(0xFFE7E2D6),
  ),
  fontFamily: 'PocketGull Sans Clinical',
  appBarTheme: const AppBarTheme(
    backgroundColor: Color(0xFFEFEAE1),
    foregroundColor: Color(0xFF292524),
    elevation: 0,
  ),
);

/// 👁️ PocketGull Obsidian Ophthalmic ThemeData (Material 3 / WCAG AAA)
final ThemeData pocketgullObsidianTheme = ThemeData(
  useMaterial3: true,
  brightness: Brightness.dark,
  scaffoldBackgroundColor: const Color(0xFF09090B),
  colorScheme: const ColorScheme.dark(
    primary: Color(0xFF38BDF8),       // Surgical Clinical Cyan
    onPrimary: Colors.black,
    secondary: Color(0xFF10B981),     // Telemetry Emerald
    onSecondary: Colors.black,
    tertiary: Color(0xFFFBBF24),      // Amber Gold
    surface: Color(0xFF0E0E14),
    onSurface: Color(0xFFF8FAFC),     // Luminous Stark White
    outline: Color(0xFF1E1E28),
  ),
  fontFamily: 'PocketGull Sans Clinical',
  appBarTheme: const AppBarTheme(
    backgroundColor: Color(0xFF0B0B0F),
    foregroundColor: Color(0xFFF8FAFC),
    elevation: 0,
  ),
);

/// Determines theme based on Circadian bio-rhythm hour (07:00 - 18:00 = Washi, 18:00 - 07:00 = Obsidian)
ThemeData getPocketGullCircadianTheme() {
  final hour = DateTime.now().hour;
  final isDaylight = hour >= 7 && hour < 18;
  return isDaylight ? pocketgullWashiTheme : pocketgullObsidianTheme;
}

File: public/brand/android-theme/compose/PocketGullTheme.kt
package com.pocketgull.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import java.util.Calendar

// 📜 Washi Rice Paper Color Scheme
val WashiLightColorScheme = lightColorScheme(
    primary = Color(0xFF0D9488),       // Celadon Bamboo Teal
    onPrimary = Color.White,
    secondary = Color(0xFFD97706),     // Persimmon Amber
    onSecondary = Color.White,
    tertiary = Color(0xFF0284C7),      // Medical Indigo
    background = Color(0xFFFAF8F2),    // Unbleached Mulberry Washi
    surface = Color(0xFFF5F2E9),
    surfaceVariant = Color(0xFFEFEAE1),
    onBackground = Color(0xFF292524),  // Sumi Charcoal Ink
    onSurface = Color(0xFF292524),
    onSurfaceVariant = Color(0xFF78716C),
    outline = Color(0xFFE7E2D6)
)

// 👁️ Obsidian Ophthalmic Color Scheme (WCAG AAA)
val ObsidianDarkColorScheme = darkColorScheme(
    primary = Color(0xFF38BDF8),       // Surgical Clinical Cyan
    onPrimary = Color.Black,
    secondary = Color(0xFF10B981),     // Telemetry Emerald
    onSecondary = Color.Black,
    tertiary = Color(0xFFFBBF24),      // Amber Gold
    background = Color(0xFF09090B),    // Deep Obsidian Void
    surface = Color(0xFF0E0E14),
    surfaceVariant = Color(0xFF14141D),
    onBackground = Color(0xFFF8FAFC),  // Luminous Stark White
    onSurface = Color(0xFFF8FAFC),
    onSurfaceVariant = Color(0xFF94A3B8),
    outline = Color(0xFF1E1E28)
)

/**
 * PocketGull Circadian Composable Theme.
 * Automatically aligns with circadian photopic daylight (07:00 - 18:00) and scotopic nightfall (18:00 - 07:00),
 * or respects system dark mode / manual override.
 */
@Composable
fun PocketGullTheme(
    circadianAuto: Boolean = true,
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit
) {
    val isNightfall = if (circadianAuto) {
        val hour = Calendar.getInstance().get(Calendar.HOUR_OF_DAY)
        hour < 7 || hour >= 18
    } else {
        darkTheme
    }

    val colorScheme = if (isNightfall) ObsidianDarkColorScheme else WashiLightColorScheme

    MaterialTheme(
        colorScheme = colorScheme,
        content = content
    )
}
